rpc/verif_proxy_test.go

```diff
@@ -47,42 +45,58 @@ func (s *ProxySuite) startRpcClient(infuraURL string) *Client {
 	require.NoError(s.T(), err)
 
 	db, close := setupTestNetworkDB(s.T())
-	defer close()
-	c, err := NewClient(gethRPCClient, 1, params.UpstreamRPCConfig{Enabled: true, URL: infuraURL}, []params.Network{}, db)
+	c, err := NewClient(gethRPCClient, 1, params.UpstreamRPCConfig{Enabled: true, URL: infuraURL}, []params.Network{}, true, db)
 	require.NoError(s.T(), err)
 
-	return c
+	return c, close
 }
 
 func (s *ProxySuite) TestRun() {
 	infuraURL := "https://mainnet.infura.io/v3/800c641949d64d768a5070a1b0511938"
-	client := s.startRpcClient(infuraURL)
+	client, closeDb := s.startRpcClient(infuraURL)
+
+	defer closeDb()
+
+	fmt.Println("Before waitForProxyHeaders")
+	ctxTimeout, _ := context.WithTimeout(context.Background(), 600*time.Second)
+	ticker := time.NewTicker(5 * time.Second)
+	for {
+		select {
+		case <-ctxTimeout.Done():
+			s.Require().Fail("Timeout reached")
+		case <-ticker.C:
+			// Let's check if handlers have been installed
+			_, found := client.handler("eth_getBalance")
+			if found {
+				fmt.Println("Proceed")
+				ticker.Stop()
+				goto proceed
+			}
+		}
+	}
 
-	// Run light client proxy
-	ctx, cancel := context.WithCancel(context.Background())
+proceed:
+	fmt.Println("after waitForProxyHeaders")
```

Comment:
Should probably be a debug log?
```suggestion
	fmt.Println("after waitForProxyHeaders")
```